### PR TITLE
Next: Added support for Shopify connector in vsf-next

### PR DESCRIPTION
**Pull request by aureate-labs-team**
### Related Issues

closes #

### Short Description and Why It's Useful

Initial release a Shopify connector for vuestorefront-next with minimum features

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/shopify/api-client/src/api/category/fetchAll.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+async function fetchAll(): Promise<Category[]> {
+  const collections = await _shopifyClient.collection
+    .fetchAll()
+    .then((collections) => {
+      return collections;
+    });
+  return collections;
+}
+
+export default fetchAll;
```
[the remark this answers is not in the document]
Comment (aureate-labs-team) [reply packages/shopify/api-client/src/api/category/fetchAll.ts:24]: We're just following the pattern which you've used for others.

File: packages/shopify/composables/src/composables/getters/cartGetters.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  return productPrice;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const getCartItemQty = (product: LineItem): number => (product as any).quantity;
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const getCartItemAttributes = (product: LineItem, filterByAttributeName?: Array<string>): any => {
```
Comment (filrak) [inline packages/shopify/composables/src/composables/getters/cartGetters.ts:51]: why `any`?

File: packages/shopify/composables/src/composables/useCart/index.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import { Cart, CartItem, Product } from '@vue-storefront/shopify-api/src/types';
+import { Coupon } from '../../types';
+import loadCurrentCart from './loadCurrentCart';
+import Cookies from 'js-cookie';
+
+export const cart: Ref<Cart> = ref(null);
+
+// @todo: implement cart
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const fetchCartLineItem = ({ currentCart, product }) => {
```
Comment (filrak) [inline packages/shopify/composables/src/composables/useCart/index.ts:15]: if its finding not fetching why its called `fetch` ?

File: packages/shopify/composables/src/composables/useCart/index.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  return currentLineItem;
+};
+
+const params: UseCartFactoryParams<Cart, CartItem, Product, Coupon> = {
+  loadCart: async () => {
+    const cartParams = {
+      checkoutId: Cookies.get('cart_id') || '',
+      id: Cookies.get('cart_id') || ''
+    };
+    const currentCart = await loadCurrentCart(cartParams);
```
Comment (filrak) [inline packages/shopify/composables/src/composables/useCart/index.ts:26]: shouldn't cookies handling happen inside api-client? Will api client work by itself without cookies handling?

File: packages/shopify/composables/src/composables/useCart/index.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    cart.value = currentCart;
```
Comment (filrak) [inline packages/shopify/composables/src/composables/useCart/index.ts:27]: please dont mutate the `cart` object (applies to all mutations in this file). It happens inside the factory based on value returned from this fn

File: packages/shopify/composables/src/composables/useCart/index.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    const cartParams = {
+      id: currentCart.id,
+      lineItems: [{ id: product.id, variantId: product.variant.id, quantity: quantity }]
+    };
+    const updatedCart = await updateCart(cartParams);
+    cart.value = updatedCart;
+    return updatedCart;
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  clearCart: async ({ currentCart }) => {
```
Comment (filrak) [inline packages/shopify/composables/src/composables/useCart/index.ts:69]: missing

File: packages/shopify/api-client/src/api/category/fetchAll.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (filrak) [reply packages/shopify/api-client/src/api/category/fetchAll.ts:24]: can you please show where it's used that way in other connectors? 

To clarify - I was mostly talking about unnecessary promise return here

[the remark this answers is not in the document]
Comment (filrak) [reply packages/shopify/api-client/src/api/category/fetchAll.ts:24]:
```
async function fetchAll(): Promise<Category[]> {
  return await _shopifyClient.collection.fetchAll()
}
export default fetchAll;
```

File: packages/shopify/composables/src/composables/useCart/index.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  applyCoupon: async ({ currentCart, coupon }) => {
+    return {updatedCart: currentCart, updatedCoupon: {}};
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  removeCoupon: async ({ currentCart }) => {
+    return {updatedCart: currentCart, updatedCoupon: {}};
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  isOnCart: ({ currentCart }) => {
```
Comment (filrak) [inline packages/shopify/composables/src/composables/useCart/index.ts:81]: missing

**Comment by piyushlathiya** [comment]
It's official now. 👍

**Pull request merged**